packages/engine-server/src/doctor/service.ts

```diff
@@ -172,6 +175,44 @@ export class DoctorService implements Disposable {
 
     let doctorAction: ((note: NoteProps) => Promise<any>) | undefined;
     switch (action) {
+      case DoctorActionsEnum.ADD_MISSING_DEFAULT_CONFIGS: {
+        const { wsRoot } = engine;
+        const rawConfig = DConfig.getRaw(wsRoot);
+        const detectResp = ConfigUtils.detectMissingDefaults({
+          config: rawConfig,
+        });
+        if (detectResp.data) {
+          const { needsBackfill, backfilledConfig } = detectResp.data;
+          if (needsBackfill) {
+            // back up dendron.yml first
+            let backupPath: string;
+            try {
+              backupPath = await DConfig.createBackup(
```

Comment:
It's probably out of scope for this PR, but we should move this to the BackupService too right?

Reply:
This is a method specific to creating backups for `dendron.yml` that already uses the `BackupService`.
Do you think these specific uses of the service should also be included in the service?

Maybe this should be in `ConfigUtils` not `DConfig`. Or maybe a separate utility class for all things backup (`BackupUtils`)?

Reply:
I must have been looking at an outdated version of `DConfig.createBackup` then, sometimes VSCode doesn't check out the right version for me. Sorry! This is fine.